### PR TITLE
Adjust the Mesos task path util

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Align the util with the latest Mesos API advancements as the "pod" task data changed which resulted in wrong paths. It also no longer necessary to iterate over the list of tasks as we already have all the bits to construct the paths.

Closes DCOS-10807

### CODE REVIEW

**Comment by greggomann** [comment]
I ran a one-off DC/OS build with this PR and it works great, thanks @orlandohohmeier! 🚢

File: src/js/utils/MesosStateUtil.js [hunk 1]
```diff
@@ -225,26 +225,25 @@ const MesosStateUtil = {
...
         .every(function (executor) {
-          // Find app/framework executor
-          if (executor != null &&
-              (executor.id === executorID || executor.id === taskID)) {
+          if (executor.id === executorID || executor.id === taskID) {
+
+            // Use the executor task path construct if it's a "pod" / TaskGroup
+            // executor (type: DEFAULT), otherwise fallback to the default
+            // app/framework behavior.
+            if (executor.type === 'DEFAULT') {
```
Comment (rcorral) [inline src/js/utils/MesosStateUtil.js]: Not a blocker for PR, but would be nice if this `DEFAULT` was extracted to a constant.

**Comment by wavesoft** [comment]
Looks good to me! I also share the same feeling with @rcorral , but I don't really think there is going to another type any time soon.

Thanks for addressing this so quickly! 💯 ✨

**Comment by orlandohohmeier** [comment]
@rcorral @wavesoft I've introduced an executor type constant, hope that's what you guys had in mind.

**Pull request merged**